### PR TITLE
[Add feature] Hide "+" button if no layers are present in the active project

### LINKED ISSUE #33: [Add feature] Hide "Add" button when project has no user-editable layers

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #33 

Steps to verify:

1. Select "example" project
2. "+" button appears as expected
3. Select "project with no layers" project
4. "+" button should no longer be visible

@gino-m   PTAL?

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java [hunk 1]
```diff
@@ -110,6 +115,25 @@
...
+        Optional.of(project)
+            .filter(Loadable::isLoaded)
+            .map(p -> containsLayers(p.value()))
+            .orElse(false);
+    addFeatureBtnVisibility.postValue(addBtnVisibility ? View.VISIBLE : View.GONE);
+  }
+
+  /** Returns true if the given project contains layers, otherwise false. */
+  private boolean containsLayers(Optional<Project> project) {
+    return project.map(p -> !p.getLayers().isEmpty()).orElse(false);
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java:130]: I think you can drop `orElse` here and call `flatMap` on line 123.

**Review by scolsen — APPROVED** [review]
Thanks, Shobhit!

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java]: Done

**Review by gino-m — CHANGES_REQUESTED** [review]
#33 asks that the button is hidden when "no user editable layers" are present. This will only make sense once we actually implement per-layer permissions as implemented as described in https://github.com/google/ground-platform/issues/228. For now, a project with no layers should never happen, since it would be useless. Can we rename where appropriate and add a TODO to make it clear that we should finish implementing this feature by checking on the layer permissions to determine whether the project has user-editable layers? Thanks again for the PR!

File: gnd/src/androidTest/java/com/google/android/gnd/ui/home/AddFeatureButtonVisibilityTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
+/*
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply gnd/src/androidTest/java/com/google/android/gnd/ui/home/AddFeatureButtonVisibilityTest.java:1]: Acknowledged

**Comment by shobhitagarwal1612** [comment]
@gino-m Can I get a re-review?

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java [hunk 3]
```diff
@@ -110,6 +115,28 @@
...
+    }
+
+    // TODO: Also check if the project has user-editable layers.
+    //  Pending feature, https://github.com/google/ground-platform/issues/228
+
+    // Project must contain at least 1 layer.
+    return project.value().map(p -> !p.getLayers().isEmpty()).orElse(false);
+  }
+
+  public LiveData<Integer> getAddFeatureBtnVisibility() {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java:136]: Nit: Btn -> Button throughout

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks! One nit, otherwise LGTM!

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java (hunk 3 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java]: Done

**Review by gino-m — APPROVED** [review]
LGTM.. Boom! 🚀

**Pull request merged**